nox/src/main.rs

```diff
@@ -115,13 +115,23 @@ fn main() -> eyre::Result<()> {
 
     let resolved_config = config.clone().resolve()?;
 
-    let (core_manager, core_manager_task) = PersistentCoreManager::from_path(
-        resolved_config.dir_config.core_state_path.clone(),
-        resolved_config.node_config.system_cpu_count,
-        resolved_config.node_config.cpus_range.clone(),
-    )?;
-
-    let core_manager: Arc<CoreManager> = Arc::new(core_manager.into());
+    let (core_manager, core_manager_task) = if resolved_config.dev_mode_config.enable {
+        let (core_manager, core_manager_task) = DevCoreManager::from_path(
+            resolved_config.dir_config.core_state_path.clone(),
+            resolved_config.node_config.system_cpu_count,
+            resolved_config.node_config.cpus_range.clone(),
+        )?;
+        let core_manager: Arc<CoreManager> = Arc::new(core_manager.into());
+        (core_manager, core_manager_task)
+    } else {
+        let (core_manager, core_manager_task) = StrictCoreManager::from_path(
+            resolved_config.dir_config.core_state_path.clone(),
+            resolved_config.node_config.system_cpu_count,
+            resolved_config.node_config.cpus_range.clone(),
+        )?;
+        let core_manager: Arc<CoreManager> = Arc::new(core_manager.into());
+        (core_manager, core_manager_task)
+    };
```

Comment:
Maybe do smth like this?
```suggestion
    let make_core_manager =  if resolved_config.dev_mode_config.enable { DevCoreManager::from_path } else { StrictCodeManager::from_path };
    let (core_manager, core_manager_task) = make_core_manager(
          resolved_config.dir_config.core_state_path.clone(),
          resolved_config.node_config.system_cpu_count,
          resolved_config.node_config.cpus_range.clone(),
      )?;
      let core_manager: Arc<CoreManager> = Arc::new(core_manager.into());

```

Reply:
Unfortunately,  it is not possible because of the types.
DevCoreManager & StrictCodeManager are different types and don't have a supertype. Here a `.into` converts them into a common enum type.